drake/doc/code_review_checklist.rst

```diff
@@ -66,14 +66,16 @@ Are you using pointers well?
 Did you document your API?  Did you comment your complexity?
 ============================================================
 
-- Every nontrivial public symbol must have a doxygen-formatted
+- Every nontrivial public symbol must have a Doxygen-formatted
   comment.
-
 - If you are uncertain of your formatting, consider
-  :ref:`generating the doxygen <documentation-generation-instructions>`
+  :ref:`generating the Doxygen <documentation-generation-instructions>`
   and checking how it looks in a browser.
+- Only use Doxygen comments (``///`` or ``/** */``) on published APIs (public
+  or protected classes and methods).  Code with private access or declared in
+  ``.cc`` files should *not* use the Doxygen format.
```

Comment:
The following link is to @sherm1's explanation about why  we adopted this convention:

https://reviewable.io/reviews/robotlocomotion/drake/2207#-KHMWRqTZ08h_aTawv6n

I've copy pasted the key part below:

> It is a lot of work to make and maintain doxygen comments and they have to be aimed at API users. Private methods are for developers eyes only and should have comments that speak to them. The doxygen feature you are describing is specifically for developers who are trying to understand a big piece of software; it isn't for API users.
> 
> Also, doxygen comments are a kind of code and must be tested and debugged. If we had them on private methods we would have to test them, or else they will be junk anyway.